### ISSUE TITLE
Disconnected data?

### ISSUE THREAD

**Issue by jhorv** [MarimerLLC/cslaforum#50]
I haven't used CSLA before, still trying to learn what it is (and isn't).  Does CSLA has any facility/provision/concept of a transient connection to data access?  I'm thinking about scenarios where the network may have a quick disconnect/reconnect cycle (like wireless/cellular roaming), and also longer disconnected periods where activity may need to be stored locally and forwarded/sync'd later?

**Comment by jonnybee** [comment]
No, but that is where the Microsoft Sync framework may be able to help you. 

CSLA basically helps you to create a business object model for working with your data, with a mindset of a ViewModel (a shape that match the UI/task). CSLA does not have any lock-in to vendors, databases etc  so you can use any type of data access you like.

**Comment by rockfordlhotka** [comment]
It is important to realize up front that CSLA is not an ORM, and it doesn't handle your database interactions - it just provides a clear/defined location for you to implement those interactions.

Entity Framework 6 has some built-in retry semantics that you can enable. I think they are primarily intended for cloud-based scenarios where a temporary disconnect may occur. Of course this would be server-side code, since EF is generally unavailable in modern client-side platforms (UWP, Android, iOS).

So really the question I suppose is whether the CSLA data portal has provisions to perform retries, and the answer is currently no. It is an interesting thought though, and something I might consider.

In terms of longer disconnects where store-and-forward would be necessary, I've considered that generally out of scope for CSLA itself. There are good data-layer technologies to handle this sort of issue, and my recommendation has always been to implement the client so it interacts with a client-side data store, and then use data-layer sync technologies to handle server communication. This is the model used (for example) by Outlook, and it is a tried-and-proven approach to solving the problem.
